Image upload and add Review

AmoghaK1 opened the pull request:
This feature branch has the following features added:
1. Home page rendering. 
2. Image upload and view feature. 
3. Adding a review feature.
4. Improved the UI of home page.
5. Added MVC Architechture for better readability.
6. Added Exception Handling.

AmoghaK1 committed:
added image uploading feature and UI

AmoghaK1 committed:
Added review feature

AmoghaK1 committed:
Converted code to MVC Architechture

cpandey05 commented on controllers/mainController.js:
```diff
@@ -0,0 +1,96 @@
+
+    res.status(201).json({message: 'Image uploaded successfully'});
+  } catch (error) {
+    res.status(500).json({ error: error.message });
```
How will this will be propagated to end user?

AmoghaK1 replied:
> How will this will be propagated to end user?

the Function called getAllImages() fetches all the images from the database and sends an array of objects containing image data to the frontend.

cpandey05 commented on controllers/mainController.js:
```diff
@@ -0,0 +1,96 @@
+    await review.save();
+    res.status(201).json({ message: 'Review added successfully' });
+  } catch (err) {
+    res.status(500).json({ error: 'Failed to add review' });
```
You should always log error for further investigation

cpandey05 commented:
There is no layered approach ..controller should not have any logic, it should be delegated to service layer.
Enitities are directly being send to the view -DTO should be used

AmoghaK1 committed:
Added service and DTO Layer

AmoghaK1 commented:
> There is no layered approach ..controller should not have any logic, it should be delegated to service layer. Enitities are directly being send to the view -DTO should be used

Added service and DTO Layer. 
Updated code with better error logs.
